model_pinocchio_vs_franka_controller:
  type: panda_torque_mpc/ModelPinocchioVsFrankaController
  arm_id: $(arg arm_id)
  urdf_path: $(find panda_torque_mpc)/config/panda_inertias_nohand.urdf

log_update_dt:
  type: panda_torque_mpc/LogUpdateDt
  arm_id: $(arg arm_id)

joint_space_ID_controller:
    type: panda_torque_mpc/JointSpaceIDController
    arm_id: $(arg arm_id)
    urdf_path: $(find panda_torque_mpc)/config/panda_inertias_nohand.urdf
    joint_names:
        - $(arg arm_id)_joint1
        - $(arg arm_id)_joint2
        - $(arg arm_id)_joint3
        - $(arg arm_id)_joint4
        - $(arg arm_id)_joint5
        - $(arg arm_id)_joint6
        - $(arg arm_id)_joint7

    # Gains for IDControl
    # Kp: 500  # Stiff
    # Kd: 50  # Stiff
    Kp: 100 
    Kd: 10 
    # Kp: 50
    # Kd: 5
    # Gains for IDControlSimplified
    kp_gains:
        - 60.0
        - 60.0
        - 60.0
        - 60.0
        - 20.0
        - 10.0
        - 5.0
    kd_gains:
        - 5.0
        - 5.0
        - 5.0
        - 2.0
        - 2.0
        - 2.0
        - 1.0
    # Trajectory simulation
    delta_q:
        - -0.3
        -  0.3
        -  0.3
        -  0.3
        -  0.3
        -  0.3
        -  0.3
    period_q:
        - 5.0
        - 5.0
        - 5.0
        - 5.0
        - 5.0
        - 5.0
        - 5.0
    publish_rate: 10.0
    # 0: IDControl
    # 1: IDControlSimplified
    # 2: PDGravity
    # 3: PureGravity
    control_variant: 0
    use_pinocchio: true
    alpha_dq_filter: 0.99 # should be close to 1 and <= 1.0
    saturate_dtau: true


task_space_ID_controller:
    type: panda_torque_mpc/TaskSpaceIDController
    arm_id: $(arg arm_id)
    urdf_path: $(find panda_torque_mpc)/config/panda_inertias_nohand.urdf
    joint_names:
        - $(arg arm_id)_joint1
        - $(arg arm_id)_joint2
        - $(arg arm_id)_joint3
        - $(arg arm_id)_joint4
        - $(arg arm_id)_joint5
        - $(arg arm_id)_joint6
        - $(arg arm_id)_joint7
    # Trajectory simulation
    delta_nu:
        - 0.07
        - 0.07
        - 0.07
        - 0.0
        - 0.0
        - 0.0
    period_nu:
        - 4.0
        - 4.0
        - 4.0
        - 5.0
        - 5.0
        - 5.0
    publish_rate: 10.0
    # 0: PosiPosture  # works
    # 1: PosePosture  # not working -> pbe in jacobian
    # 2: TSID         # TODO
    control_variant: 2
    use_pinocchio: true
    alpha_dq_filter: 1.0 # should be close to 1 and <= 1.0
    # Controller configs
    # kp_ee: 500  # Stiff
    # kd_ee: 100 
    kp_ee: 100 
    kd_ee: 20 
    # kp_ee: 20
    # kd_ee: 5 
    kp_q: 100
    kd_q: 40
    w_ee: 1
    w_q: 0.01
    tau_limit_scale: 0.01
    v_limit_scale: 0.1
    tsid_ee_mask: [1,1,1,1,1,1]
